src/recipe/s3.service.ts

```diff
@@ -14,16 +21,46 @@ export class S3Service {
 
   constructor(private configService: ConfigService) {}
 
-  async uploadFileToS3(bucket: string, fileName: string, file: Buffer) {
+  async generatePresignedUrl(
+    bucket: string,
+    key: string,
+    command: GetSignedUrlCommand,
+  ) {
     const params = {
       Bucket: bucket,
-      Key: fileName,
-      Body: file,
+      Key: key,
     };
-    const command = new PutObjectCommand(params);
-    await this.s3Client.send(command);
-    return `https://${bucket}.s3.${this.configService.getOrThrow(
-      'AWS_S3_REGION',
-    )}.amazonaws.com/${fileName}`;
+    if (command === GetSignedUrlCommand.PUT) {
+      return getSignedUrl(this.s3Client, new PutObjectCommand(params), {
+        expiresIn: 60,
+      });
+    } else if (command === GetSignedUrlCommand.GET) {
+      return getSignedUrl(this.s3Client, new GetObjectCommand(params), {
+        expiresIn: 60,
+      });
+    }
+  }
+
+  async uploadFileToS3(bucket: string, key: string, file: Buffer) {
+    const signedUrl = await this.generatePresignedUrl(
```

Comment:
you do not have to use pre-signed URLs to upload files to s3. You already had a working setup for this